### ISSUE TITLE
configuration-file conflicts on upgrade

### ISSUE THREAD

**Issue by hex-m** [bigbluebutton/bbb-install#330]
I witnessed multiple times, that an upgrade with bbb-install.sh warned about conflicts in configuration files.


For example with the upgrade to 2.2.31 I got:

```
Setting up bbb-config (1:2.2.0-225) ...                                                                                                                   
                                                                                                                                                          
Configuration file '/etc/systemd/system/bbb-apps-akka.service.d/override.conf'                                                                            
 ==> File on system created by you or by a script.                                                                                                        
 ==> File also in package provided by package maintainer.                                                                                                 
   What would you like to do about it ?  Your options are:                                                                                                
    Y or I  : install the package maintainer's version                                                                                                    
    N or O  : keep your currently-installed version                                                                                                       
      D     : show the differences between the versions                                                                                                   
      Z     : start a shell to examine the situation                                                                                                      
 The default action is to keep your current version.                                                                                                      
*** override.conf (Y/I/N/O/D/Z) [default=N] ? dpkg: error processing package bbb-config (--configure):                                                    
 end of file on stdin at conffile prompt                                                                                                                  
Errors were encountered while processing:                                                                                                                 
 bbb-config                                                                                                                                               
E: Sub-process /usr/bin/dpkg returned an error code (1)

```

Every occurrence of `apt-get install` and `apt-get dist-upgrade` should specify a strategy for how to deal with configuration conflicts.

`-o DPkg::options::="--force-confdef" -o DPkg::options::="--force-confold"`

**Comment by lonesomewalker** [comment]
Can't replicate that on multiple thousand instances.
Sure, if there are modifications made in config files you get this dialog, but NOT when using defaults so far.

**Comment by hex-m** [comment]
Interesting - thank you for the comment! I have no idea what could have changed those configuration files on my servers that would lead to those conflicts. I will do a more detailed analysis at the next update.

**Comment by lonesomewalker** [comment]
The question is:
did you change something manually?

In general these interupts should only happen if the default settings have been modified manually and so they differ from the packaged ones.

**Comment by hex-m** [comment]
I didn't change anything manually but I think I found the [commit](https://github.com/bigbluebutton/bbb-install/commit/323bc02321f362ac2b41de033a10b08c062d79fb) that was responsible for the change in those `override.conf` files. There is also another [commit](https://github.com/bigbluebutton/bbb-install/commit/e3d8e267e2bbcae6d8c9cce467138afec4a3956f) where `bbb-install` changed `WebRtcEndpoint.conf.ini` instead of the change coming with a new package-version. So everyone who uses `bbb-install` should have gotten those errors afaiu.

**Comment by ffdixon** [comment]
The `override.conf` files were a bit of a hack because I introduced them in the packaging script for `bbb-config` and in `bbb-install.sh, but didn't make them exactly the same so hence the prompts.

I've since removed them from the post install script for `bbb-config`, but they remain in `bbb-install.sh`, see

 https://github.com/bigbluebutton/bbb-install/blob/master/bbb-install.sh#L365

Our intent is to add these override scripts to the packaging themselves, so the above `bbb-install.sh` is temporary.

> There is also another commit where bbb-install changed WebRtcEndpoint.conf.ini instead of the change coming with a new package-version. So everyone who uses bbb-install should have gotten those errors afaiu.

This [commit](https://github.com/bigbluebutton/bbb-install/commit/e3d8e267e2bbcae6d8c9cce467138afec4a3956f) was to get away from using FreeSWITCH stun server, which was very unreliable.  In looking at this, a better approach would be to just change the `stunServerAddress` and `stunServerPort` if it still points to the old FreeSWITCH stun server.
